### ISSUE TITLE
NDA token generator uses depreciated interface

### ISSUE THREAD

**Issue by dnkennedy** [datalad/datalad#3674]
#### What is the problem?
NDA-tools have a new version of a script to obtain token from NDA. Here is some traceback from datalad showing it using old URL.

```
(Pdb) p response
b'<html>\n<head><title>\nWeb Services\n</title></head>\n<body>\n<h1>Web Services</h1>\n<table width=\'100%\' border=\'1\'>\n<tr>\n<td>\nEndpoint\n</td>\n<td>\nInformation\n</td>\n</tr>\n<tr>\n<td>\n<table border="0"><tr><td>Service Name:</td><td>{http://dataManagerService}dataManagerServicePorts</td></tr><tr><td>Port Name:</td><td>{http://dataManagerService}dataManagerService</td></tr></table>\n</td>\n<td>\n<table border="0"><tr><td>Address:</td><td>https://nda.nih.gov:443/DataManager/dataManager</td></tr><tr><td>WSDL:</td><td><a href="https://nda.nih.gov:443/DataManager/dataManager?wsdl">https://nda.nih.gov:443/DataManager/dataManager?wsdl</a></td></tr><tr><td>Implementation class:</td><td>gov.nih.ndar.ws.datamanager.DataManagerServer</td></tr></table>\n</td>\n</tr>\n</table>\n</body>\n</html>\n'
(Pdb) up
> /Users/davidkennedy/miniconda3/lib/python3.7/site-packages/datalad/support/third/nda_aws_token_generator.py(82)__make_request()
-> return self.__parse_response(response.read())
(Pdb) l
 77  	        }
 78  	
 79  	        request = urllib_request.Request(self.url, data=request_message, headers=headers)
 80  	        logging.debug(request)
 81  	        response = urllib_request.urlopen(request)
 82  ->	        return self.__parse_response(response.read())
 83  	
 84  	    def __parse_response(self, response):
 85  	        logging.debug('parsing response')
 86  	        tree = etree.fromstring(response)
 87  	
(Pdb) p self.url
'https://ndar.nih.gov/DataManager/dataManager'
(Pdb) up
> /Users/davidkennedy/miniconda3/lib/python3.7/site-packages/datalad/support/third/nda_aws_token_generator.py(34)generate_token()
-> return self.__make_request(request_xml)
```

**Comment by obenshaindw** [comment]
@yarikoptic I think this may actually have to do with the rebrand work that was done, which effectively changed the domain name for all of our services from https://ndar.nih.gov to https://nda.nih.gov.  Requests to https://ndar.nih.gov/DataManager/dataManager return a 301 redirect with the location header set to https://nda.nih.gov/DataManager/dataManager.

@dnkennedy was referring to [nda-tools](https://github.com/NDAR/nda-tools/blob/master/NDATools/Download.py#L113) where we have provided a client script that fetches a list of files contained in an NDA package from the DataManager service and proceeds to then download those files, or optionally will accept a s3 URL as a command-line argument, and uses a DataManager provided token to authenticate/authorize the downloads from S3. If you haven't looked at this yet, it might be useful while looking at the DataLad component you authored for interfacing with NDA.

**Comment by yarikoptic** [comment]
I believe it was fixed by #4824 (and may be some follow ups)
